sources/packages/api/src/route-controllers/student/student.controller.ts

```diff
@@ -579,4 +579,47 @@ export class StudentController extends BaseController {
       hasRestriction: studentRestrictionStatus.hasRestriction,
     } as StudentDetailDTO;
   }
+
+  /**
+   * controller to save the files submitted by the student
+   * with student uploader form.
+   * @Body payload
+   */
+  @AllowAuthorizedParty(AuthorizedParties.student)
+  @Post("upload-files")
+  async saveStudentUploadedFiles(
+    @UserToken() userToken: IUserToken,
+    @Body() payload: StudentFileUploaderDto,
+  ): Promise<void> {
+    const existingStudent = await this.studentService.getStudentByUserId(
+      userToken.userId,
+    );
+    if (!existingStudent) {
+      throw new NotFoundException(
+        `No student was found with user id ${userToken.userId}`,
+      );
+    }
+    if (payload.submittedForm.applicationNumber) {
+      // Here we are checking the existence of an application irrespective of its status
+      const validApplication =
+        await this.applicationService.getApplicationByApplicationNumberStudent(
+          payload.submittedForm.applicationNumber,
+          existingStudent,
+        );
+
+      if (!validApplication) {
+        throw new NotFoundException(
```

Comment:
This would be more an "unprocessable exception" or "bad request exception" since the URI was found and just the data is not possible to be processed due to an invalid parameter.